services/builder/configProcessor.go

```diff
@@ -216,3 +222,54 @@ func mergeDeleteCounts(dst, src map[string]*nodeCount) map[string]*nodeCount {
 	}
 	return dst
 }
+
+// prepareConfigToDelete function queries for cluster those needs ro be deleted from current state.
+// It also updated the config object to remove the clusters to be deleted from current state.
+// returns *pb.Config which contains clusters (both k8s and lb) that needs to be deleted.
+func prepareConfigToDelete(config *pb.Config) *pb.Config {
+	configToDelete := proto.Clone(config).(*pb.Config)
+	var k8sClustersToDelete, remainingCsK8sClusters []*pb.K8Scluster
+	var LbClustersToDelete, remainingCsLbClusters []*pb.LBcluster
+
+OuterK8s:
+	for _, csCluster := range config.CurrentState.Clusters {
+		for _, dsCluster := range config.DesiredState.Clusters {
+			if isEqual(dsCluster.ClusterInfo, csCluster.ClusterInfo) {
+				remainingCsK8sClusters = append(remainingCsK8sClusters, csCluster)
+				continue OuterK8s
+			}
+		}
+		k8sClustersToDelete = append(k8sClustersToDelete, csCluster)
+	}
+OuterLb:
+	for _, csLbCluster := range config.CurrentState.LoadBalancerClusters {
+		for _, dsLbCluster := range config.DesiredState.LoadBalancerClusters {
+			if isEqual(dsLbCluster.ClusterInfo, csLbCluster.ClusterInfo) {
+				remainingCsLbClusters = append(remainingCsLbClusters, csLbCluster)
+				continue OuterLb
+			}
+		}
+		LbClustersToDelete = append(LbClustersToDelete, csLbCluster)
+	}
+	configToDelete.CurrentState.Clusters = k8sClustersToDelete
+	configToDelete.CurrentState.LoadBalancerClusters = LbClustersToDelete
+	updateCurrentState(config, remainingCsK8sClusters, remainingCsLbClusters)
+	return configToDelete
+}
+
+// isEqual function checks if the two cluster from desiredState and Current state are same by comparing
+// names and hashes
+// return boolean value, True if the match otherwise False
+func isEqual(dsClusterInfo, csClusterInfo *pb.ClusterInfo) bool {
+	if dsClusterInfo.Name == csClusterInfo.Name && dsClusterInfo.Hash == csClusterInfo.Hash {
+		return true
+	}
+	return false
+}
+
+// updateCurrentState function will update the Clusters and LoadBalancerClusters for the config passed.
+// the function has side effects since it updated the config which is passed as a pointer reference
+func updateCurrentState(config *pb.Config, k8sCluster []*pb.K8Scluster, lbClusters []*pb.LBcluster) {
+	config.CurrentState.Clusters = k8sCluster
+	config.CurrentState.LoadBalancerClusters = lbClusters
+}
```

Comment:
Ditto. Do we need this as a separate function? It's only used in one place and the body of the function are 2 assignments, It would be better to be inlined. i.e. don't abstract it in a function.

Reply:
We can compromise; don't use the function and just add a comment about it so the behaviour is documented